db/db_impl/db_impl.cc

```diff
@@ -1119,8 +1119,16 @@ Status DBImpl::SetDBOptions(
   WriteContext write_context;
   {
     InstrumentedMutexLock l(&mutex_);
+    bool is_changed;
     s = GetMutableDBOptionsFromStrings(mutable_db_options_, options_map,
-                                       &new_options);
+                                       &new_options, &is_changed);
+    if (!is_changed) {
```

Comment:
See the note below about making this a new routine.  Would allow for the sanitization in the following lines to take place before the comparison.

Reply:
removed.